File: examples/todo-java-spring/output/src/main/java/com/example/todo/PriorityTaskRepository.java
package com.example.todo;

import org.springframework.stereotype.Repository
import org.springframework.data.jpa.repository.JpaRepository
import java.util.List
import java.util.UUID
import java.time.LocalDate
@Repository
public interface PriorityTaskRepository extends JpaRepository<PriorityTask, java.util.UUID> {
    java.util.List<PriorityTask> findByCompleted(boolean completed);
    java.util.List<PriorityTask> findByDueDateBefore(java.time.LocalDate date);
    java.util.List<PriorityTask> findByPriority(String priority);
    java.util.List<PriorityTask> findByCompletedAndPriority(boolean completed, String priority);
}

File: examples/todo-kotlin-quarkus/output/src/main/kotlin/com/example/todo/PriorityTaskRepository.kt
package com.example.todo

import jakarta.enterprise.context.ApplicationScoped
import io.quarkus.hibernate.orm.panache.PanacheRepository
@ApplicationScoped
open class PriorityTaskRepository : PanacheRepository<PriorityTask> {
    fun findByCompleted(completed: Boolean): java.util.List<PriorityTask> {
        return find("completed", completed).list()
    }
    fun findByDueDateBefore(date: java.time.LocalDate): java.util.List<PriorityTask> {
        return find("dueDate < ?1", date).list()
    }
    fun findByPriority(priority: String): java.util.List<PriorityTask> {
        return find("priority", priority).list()
    }
}